lib/std/fs/path.zig

```diff
@@ -1182,3 +1182,69 @@ fn testRelativeWindows(from: []const u8, to: []const u8, expected_output: []cons
     defer testing.allocator.free(result);
     testing.expectEqualSlices(u8, expected_output, result);
 }
+
+/// Returns the extension of the file name (if any).
+/// This function will search for the file extension (separated by a `.`) and will return the text after the `.`.
+/// Files that end with `.` are considered to have no extension, files that start with `.`
+/// Examples:
+/// - `"main.zig"`     ⇒ `"zig"`
+/// - `"src/main.zig"` ⇒ `"zig"`
+/// - `".gitignore"`   ⇒ `null`
+/// - `"keep."`        ⇒ `null`
+/// - `"src.keep.me"`  ⇒ `"me"`
+pub fn extension(path: []const u8) ?[]const u8 {
+    const filename = basename(path);
+    return if (std.mem.lastIndexOf(u8, filename, ".")) |index|
+        if (index == 0 or index == filename.len - 1)
+            null
+        else
+            filename[index + 1 ..]
+    else
+        null;
+}
+
+fn testExtension(path: []const u8, expected: ?[]const u8) void {
+    const actual = extension(path);
+
+    if (expected) |must_be| {
+        std.testing.expect(actual != null);
+        std.testing.expectEqualStrings(must_be, actual.?);
+    } else {
+        std.testing.expectEqual(expected, actual);
+    }
+}
+
+test "extension" {
+    testExtension("", null);
+    testExtension(".", null);
+    testExtension("a.", null);
+    testExtension("abc.", null);
+    testExtension(".a", null);
+    testExtension(".file", null);
+    testExtension(".gitignore", null);
+    testExtension("file.gitignore", "gitignore");
```

Comment:
sort of a weird test case (gitignore that is), maybe just `file.ext`?

Reply:
Why are we shying away from eccentricity

Reply:
> sort of a weird test case (gitignore that is), maybe just file.ext?

I am a lazy copy-paster. If you want, i can still adjust that  :)